### PR TITLE
Example for PlantUML diagrams without Graphviz dot installation

### PR DESCRIPTION
**Pull request by kriegaex**
Hello.

I do not expect you to merge this as-is, feel free to
  * adjust the examples to your needs,
  * also update the [Asciidoctor docs for the PlantUML plugin](https://docs.asciidoctor.org/diagram-extension/latest/diagram_types/plantuml/) to mention the [Smetana engine](https://plantuml.com/de/smetana02), see also the [former Smetana repository](https://github.com/plantuml/smetana?tab=readme-ov-file) (now merged into PlantUML proper).

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -43,15 +43,6 @@ jobs:
           path: ~/.m2/repository
           key: mvn-cache-${{ runner.os }}-${{ env.java_version }}
           restore-keys: mvn-cache-${{ runner.os }}-${{ env.java_version }}
-      - name: Install Graphviz (Linux)
```
Comment (abelsromero) [inline .github/workflows/build.yaml:46]: I'd like to keep this as a self-documenting requirement. Not all distro come with it by default and it's one of the typical initial pitfalls users suffer.

**Comment by abelsromero** [comment]
> I do not expect you to merge this as-is, feel free to

I do in fact expect to be able to get any contribution in with the original authoring and attributions :smile: 
I added a single comment, if we can get that the PR goes it as-is.



> * also update the [Asciidoctor docs for the PlantUML plugin](https://docs.asciidoctor.org/diagram-extension/latest/diagram_types/plantuml/) to mention the [Smetana engine](https://plantuml.com/de/smetana02)

You can mention it in the repo or do the change yourself, they'll be happy to review. Don't take it wrong, we are distributed teams and I am focused on the maven/java side of things and don't use 'diagram' myself. You'll be able to provide a better description than me. The exact file is this one https://github.com/asciidoctor/asciidoctor-diagram/blob/master/docs/modules/ROOT/pages/diagram_types/plantuml.adoc

**Review by abelsromero — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (kriegaex) [reply .github/workflows/build.yaml:46]: Absolutely, keep it. Do not merge. Feel free to close this PR anytime you think you refreshed the sampples and docs the way you prefer.

Like I said, it was meant for showcasing that it works without Graphviz for simple cases, so you can decide how you want to document it or if you want to add another example project specifically for this simple case.

Comment (kriegaex) [reply .github/workflows/build.yaml:46]: > PS: I admit I am surprised the test pass without installing dot`, is not the case in my local machine 🤔 do you know why?

No, not really. I actually have a local Graphviz on my Windows path locally, I just renamed the directory for the test. Did you do that too, to make sure that the test cannot find it? Maybe it finds a local installation, and it is too old or whatever. But this is a wild guess, I really do not know. I use PlantUML approximately once a year.

**Comment by kriegaex** [comment]
> You can mention it in the repo or do the change yourself, they'll be happy to review. Don't take it wrong, we are distributed teams

But at least, probably you know each other.

> and I am focused on the maven/java side of things

Well, Smetana is a Java port of (a subset of) the original code.

> and don't use 'diagram' myself. You'll be able to provide a better description than me. The exact file is this one https://github.com/asciidoctor/asciidoctor-diagram/blob/master/docs/modules/ROOT/pages/diagram_types/plantuml.adoc

Don't get me wrong either, but for the last two weeks or so I have spent more time creating issues and PRs in several OSS projects, instead of getting anything done in my own project, just because I am too nice. I need to focus, too.

BTW, if you know a lot about the Maven and Java side of things, can you happen to answer [this Zulip question](https://asciidoctor.zulipchat.com/#narrow/stream/279642-users/topic/Use.20PlantUML.20in.20Maven.20Site/near/419581942)?

File: .github/workflows/build.yaml (hunk 1 above)
Comment (abelsromero) [reply .github/workflows/build.yaml:46]: In my machine it's consistent, I run Manjaro and if I uninstall the [graphviz package](https://archlinux.org/packages/extra/x86_64/graphviz/) which I validated contains the `dot` binary, `[plantuml, dot-example]` diagram fails. But in CI, it seems to be working, maybe Ubuntu has some `dot` but I doubt Windows and Mac do too. I'll make a note to see if I can find some time next week.

**Comment by abelsromero** [comment]
Ported changes to https://github.com/asciidoctor/asciidoctor-maven-examples/pull/245

**Pull request closed without merging**